pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -889,11 +893,255 @@ func removeNodeIPAddressesFromBackendPool(
+
+	if lb.LoadBalancerPropertiesFormat == nil || lb.LoadBalancerPropertiesFormat.BackendAddressPools == nil {
+		return nil, nil
+	}
+
+	backendPrivateIPv4s, backendPrivateIPv6s := utilsets.NewString(), utilsets.NewString()
+	for _, bp := range *lb.BackendAddressPools {
+		found := lbBackendPoolNames.Has(pointer.StringDeref(bp.Name, ""))
+		if found {
+			klog.V(10).Infof("bpi.GetBackendPrivateIPs for service (%s): found wanted backendpool %s", serviceName, pointer.StringDeref(bp.Name, ""))
+			if bp.BackendAddressPoolPropertiesFormat != nil && bp.LoadBalancerBackendAddresses != nil {
+				for _, backendAddress := range *bp.LoadBalancerBackendAddresses {
+					ipAddress := backendAddress.IPAddress
+					if ipAddress != nil {
+						klog.V(2).Infof("bpi.GetBackendPrivateIPs for service (%s): lb backendpool - found private IP %q", serviceName, *ipAddress)
+						if utilnet.IsIPv4String(*ipAddress) {
+							backendPrivateIPv4s.Insert(*ipAddress)
+						} else if utilnet.IsIPv6String(*ipAddress) {
+							backendPrivateIPv6s.Insert(*ipAddress)
+						}
+					} else {
+						klog.V(4).Infof("bpi.GetBackendPrivateIPs for service (%s): lb backendpool - found null private IP", serviceName)
+					}
+				}
+			}
+		} else {
+			klog.V(10).Infof("bpi.GetBackendPrivateIPs for service (%s): found unmanaged backendpool %s", serviceName, pointer.StringDeref(bp.Name, ""))
+		}
+	}
+	return backendPrivateIPv4s.UnsortedList(), backendPrivateIPv6s.UnsortedList()
+}
+
+func (bpi *backendPoolTypePodIP) ReconcileBackendPools(_ string, service *v1.Service, lb *network.LoadBalancer) (bool, bool, *network.LoadBalancer, error) {
+	var newBackendPools []network.BackendAddressPool
+	if lb.BackendAddressPools != nil {
+		newBackendPools = *lb.BackendAddressPools
+	}
+
+	var backendPoolsUpdated, shouldRefreshLB bool
+	foundBackendPools := utilsets.NewString()
+	lbName := *lb.Name
+	serviceName := getServiceName(service)
+
+	endpointSliceList, er := bpi.getEndpointSliceListForService(service)
+
+	if er != nil {
+		klog.Errorf("bpi.ReconcileBackendPools: failed to get endpoint slice list for service %q, error: %s", service.Name, er.Error())
+		return false, false, nil, er
+	}
+
+	lbBackendPoolNames := bpi.getAllBackendPoolNamesForEndpointSliceList(service, endpointSliceList)
+	// bp is never preconfigured in case of pods
+	isBackendPoolPreConfigured := false
+
+	var (
+		err error
+	)
+	for i := len(newBackendPools) - 1; i >= 0; i-- {
+		bp := newBackendPools[i]
+		found := lbBackendPoolNames.Has(pointer.StringDeref(bp.Name, ""))
+		if found {
+			klog.V(10).Infof("bpi.ReconcileBackendPools for service (%s): found wanted backendpool. Not adding anything", serviceName)
+			foundBackendPools.Insert(*bp.Name)
+		} else {
+			klog.V(10).Infof("bpi.ReconcileBackendPools for service (%s): found unmanaged backendpool %s", serviceName, *bp.Name)
+			//TODO: (anujbansal) Uncomment and prevent deletion of default backend pools (kubernetes and aksOutboundBackendPool)
```

Comment:
As discussed, coming up unique prefix-name per service will prevent any issue with Remove.

Reply:
We have used the backend pools in the lb rules to resolve this. The lb rule name will have the service uid to identify the service it belongs to.